pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -413,26 +410,19 @@ func (r *ReconcileHyperConverged) ensureHcoDeleted(req *hcoRequest) (reconcile.R
 			// to be able to clearly distinguish between an explicit
 			// refuse from other operator and any other kind of error that
 			// could potentially happen in the process
+
 			errT := ErrHCOUninstall
 			errMsg := uninstallHCOErrorMsg
-			switch i {
-			case 0:
+			switch obj.(type) {
+			case *kubevirtv1.KubeVirt:
 				errT = ErrVirtUninstall
 				errMsg = uninstallVirtErrorMsg + err.Error()
-			case 1:
+			case *cdiv1alpha1.CDI:
 				errT = ErrCDIUninstall
 				errMsg = uninstallCDIErrorMsg + err.Error()
 			}
```

Comment:
why there are no `case` statements for CNAO and KvCTB?

Reply:
We don't expect delete rejection from them